### ISSUE TITLE
Record history of user logins to a table in the database

### ISSUE THREAD

**Issue by entrotech** [hackforla/food-oasis#834]
### Overview
The new redesign of the Volunteer Dashboard calls for displaying to a user their most recent previous login.  We need to add a table to record the login history and modify the login process to insert a record to that table each time the user logs in.

The needs: 
- Volunteers sometimes ask to get written/emailed proof of the hours they've volunteered from the volunteering organization, so this tool would allow the volunteer manager to easily look up logins to know what the sessions and estimate hours worked.  (since much of the work is done on the phone, might not have actual records of when the volunteer stopped working)
  - Stretch: Does the have some record of volunteer interactions with the system that could give an idea of volunteering end-time?
- Volunteer coordinators would like to have a list of currently-signed in volunteers so that they can assign tasks to those present.  Currently, this is seems to be done by hand: by looking at names of participants on a zoom call, matching those up to volunteer logins, and going from there?  

### Action Items
- [x] Confirm that we need this (or parts of this) with volunteer coordinators
- [x] Sketch out MVP UI design
- [x] Put together MVP (see below) that shows timestamped log-ins which can be filtered by username.  Show Jordan.
- [x] Does this meet Jordan or other team member needs?  Anything missing (and hopefully easy)?
- [ ] Implement
- [ ] What workarounds to volunteer coordinators do now?
- [ ] TBD

### Resources/Instructions
We want a complete hsitory of logins, rather than just the most recent prior login, so we use that to possibly generate reports on volunteer participation.
Also, need to determine behavior when browser remembers sessions between visits (maybe session timeout should be shortened?)

**Comment by entrotech** [comment]
Don't start work on this until we finish #896, and confirmm that we need it first.

**Comment by fancyham** [comment]
A workaround that might work is to display a single page that shows recent sign-in logs, with a filterable log history of say, the past 100 records.

Nice-to-have would be to select a time period (date-date, year, and/or past X weeks/months) that works with the filter.

- [ ] Log display, showing recent log-ins: Date and time, username.  
- [ ] Filter by typing username (stretch goal: autocomplete for usernames)
- [ ] Check with Jordan (volunteer coordinator) for most useful date ranges and usefulness.

Stretch goals
- [ ] Nice to have: Show how long user was active, perhaps estimating based on last interaction with server + 5 minutes.
- [ ] Nice to have Version A: User can pick a date range (date to date).  Default is today only.
- [ ] Nice to have Version B: User can pick a date range ("Today", "Last 30 days", "Past 6 months", or date range) as well as date range.  Default is "Today"
- [ ] Nice to have: items from a specific date are grouped together
- [ ] 

![#834 Display log-in times, log out times (activity)](https://user-images.githubusercontent.com/3376957/131943650-f6f10334-67f1-4a82-9d4c-792c83eb072f.png)

**Comment by entrotech** [comment]
The MVP version is merged on PR #1136. At this point, it is ready for @gigicobos and Jordan Smith and others to review and suggest future direction.

**Comment by gigicobos** [comment]
@entrotech Working with LA works to set up a meeting and review the feature with them (Jordan, Stephanie, and Charles). Not sure if we'd be able to meet before the holidays but I'll keep the team posted with comments. Thank you very much!

**Comment by Benbaillou** [comment]
Note for myself:

New feature to add: Count nb of organisations done by volunteers
- Goal: track the progress of each volunteer
- Why:ensure volunteers complete their tasks in order for organisations to be verified and that volunteers complete their objectives
- How: count the number of organisations checked by volunteers after pressing CLA button "hand off" or "Submitted"

**Comment by Benbaillou** [comment]
Hi @gigicobos, @entrotech @fancyham - please find above the last updates on this issue and the improvements needed based on Jordan's feedback.

Please provide your comments.

Thanks,

**Comment by Benbaillou** [comment]
@souronion - here is an issue I updated. I would love your design comments on this one.

**Comment by fancyham** [comment]
Hi folks, is this currently in progress?

**Comment by entrotech** [comment]
@Erin @staceyrebekahscott I think we need to revisit this issue, and decide how to move forward.  There is a basic login history available to the Admins now, but do we need additional functionality?

**Comment by staceyrebekahscott** [comment]
@itserindean Moving this to the Product board. As you have more involved with this I will defer to your judgement as to what else is needed here, or if we should close this as completed.

**Comment by itserindean** [comment]
I don't think we need additional functionality so i'm closing as completed
